python/langsmith/client.py

```diff
@@ -5269,9 +5269,15 @@ def pull_prompt_commit(
         owner, prompt_name, commit_hash = ls_utils.parse_prompt_identifier(
             prompt_identifier
         )
-        use_optimization = ls_utils.is_version_greater_or_equal(
-            self.info.version, "0.5.23"
-        )
+        try:
+            use_optimization = ls_utils.is_version_greater_or_equal(
+                self.info.version, "0.5.23"
+            )
+        except ValueError:
+            logger.exception(
+                "Failed to parse LangSmith API version. Defaulting to using optimization."
+            )
+            use_optimization = True
```

Comment:
Gonna check with @langchain-infra on this one. I think we have very few deployments out < 0.5.23, meaning that more positive EV from defualting to optimized